code/modules/wiremod/port.dm

```diff
@@ -239,44 +220,7 @@
  * Arguments:
  * * port_to_register - The port to connect the input port to
  */
-/datum/port/input/proc/set_input(new_value, send_update = TRUE)
-	if(isatom(input_value))
-		UnregisterSignal(input_value, COMSIG_PARENT_QDELETING)
-	input_value = convert_value(input_value, new_value)
-	if(isatom(input_value))
-		RegisterSignal(input_value, COMSIG_PARENT_QDELETING, .proc/null_output)
-
-	SEND_SIGNAL(src, COMSIG_PORT_SET_INPUT, input_value)
-	if(trigger && send_update)
+/datum/port/input/put(v)
+	..(v)
+	if(trigger)
 		TRIGGER_CIRCUIT_COMPONENT(connected_component, src)
-
-/// Signal handler proc to null the input if an atom is deleted. An update is not sent because this was not set by anything.
-/datum/port/input/proc/null_output(datum/source)
-	SIGNAL_HANDLER
-	if(input_value == source)
-		input_value = null
-
-/datum/port/input/disconnect()
-	unregister_output_port()
-	return ..()
-
-/datum/port/input/set_datatype(type_to_set)
-	. = ..()
-	set_input(default)
-
-/datum/port/input/proc/unregister_output_port()
-	SIGNAL_HANDLER
-	if(!connected_port)
-		return
-	UnregisterSignal(connected_port, list(
-		COMSIG_PARENT_QDELETING,
-		COMSIG_PORT_SET_OUTPUT,
-		COMSIG_PORT_DISCONNECT
-	))
-	connected_port = null
-	set_input(default)
```

Comment:
re-add all of these signal handlers and attached them to their appropriate signals

Reply:
`COMSIG_PARENT_QDELETING` isn't necessary since `Destroy` disconnects all.